lib/systeminfo/dns_test.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright 2018 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package systeminfo
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/gravitational/gravity/lib/storage"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestResolvParser(t *testing.T) {
+	var tests = []struct {
+		in  string
+		out *storage.ResolvConf
+	}{
+		{
+			`
+			domain domain.com
+			nameserver 1.1.1.1
+			`,
+			&storage.ResolvConf{
+				Ndots:    1,
+				Timeout:  5,
+				Attempts: 2,
+				Domain:   "domain.com",
+				Search:   []string{"domain.com"},
+				Servers:  []string{"1.1.1.1"},
+			},
+		},
+		{
+			`
+			# domain domain.com
+			nameserver 1.1.1.1
+			`,
+			&storage.ResolvConf{
+				Ndots:    1,
+				Timeout:  5,
+				Attempts: 2,
+				Servers:  []string{"1.1.1.1"},
+			},
+		},
+		{
+			`
+			# domain domain.com
+			nameserver 1.1.1.1`, // <- missing newline at EOF
```

Comment:
the comment could be a literal to use with `assert.Nil(t, err, tt.comment)`:
```go
	`
	# domain domain.com
	nameserver 1.1.1.1`,
	comment: "missing newline at EOF",
```

Reply:
It could... I don't think I would remove the comment pointing at the character indicating it should be missing a newline at EOF though, as it makes it very clear that it shouldn't be reformatted to match the other blocks, and if someone does reformat it by accident, it won't cause the test to begin failing.